src/hotspot/cpu/riscv/macroAssembler_riscv.cpp

```diff
@@ -1625,21 +1646,18 @@ void MacroAssembler::kernel_crc32(Register crc, Register buf, Register len,
     beqz(len, L_exit);
 
     subw(len, len, 1);
-    lwu(tmp1, Address(buf));
-    andi(tmp2, tmp1, right_8_bits);
-    update_byte_crc32(crc, tmp2, table0);
+    lbu(tmp1, Address(buf));
```

Comment:
I guess after the alignment prelogue added in this pr, the previous tail implementation should be fine now.
As in the case of java heap or native heap, the tail of passed-in buf can not cross the boundary of pages.

Reply:
Yeah, maybe. But I don't think it's good to do things that would exceed buffer limit :-)
I prefer a safer approach which won't performance much because it's tail processing.